### PR TITLE
feat(b-tabs): add ability to provide custom tab button attributes (closes: #4803)

### PR DESCRIPTION
**Pull request by andrewvasilchuk**
### Describe the PR

I need somehow to assign custom attributes to the rendered button.

Closes: #4803

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [x] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples`, `chore(docs): fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] Includes any needed TypeScript declaration file updates
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] CodeCov for patch has met target
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

Issue: #4803

### CODE REVIEW

**Comment by tmorehouse** [comment]
Could you provide us with an example use case for this?

Before creating PR's it is best to create an issue explaining the feature request and an example use case.  That way, if a PR is accepted we can track it against an issue.

File: src/components/tabs/tabs.js [hunk 1]
```diff
@@ -114,16 +131,7 @@ const BTabButtonHelper = /*#__PURE__*/ Vue.extend({
...
-          role: 'tab',
-          id: this.id,
-          // Roving tab index when keynav enabled
-          tabindex: this.tabIndex,
-          'aria-selected': this.tab.localActive && !this.tab.disabled ? 'true' : 'false',
-          'aria-setsize': this.setSize,
-          'aria-posinset': this.posInSet,
-          'aria-controls': this.controls
-        },
+        attrs: linkAttrs,
```
Comment (tmorehouse) [inline src/components/tabs/tabs.js]: Just simply do:

```js
        attrs: {
          ...tab.titleLinkAttributes,
          role: 'tab',
          id: this.id,
          // Roving tab index when keynav enabled
          tabindex: this.tabIndex,
          'aria-selected': this.tab.localActive && !this.tab.disabled ? 'true' : 'false',
          'aria-setsize': this.setSize,
          'aria-posinset': this.posInSet,
          'aria-controls': this.controls
        },
```

Comment (tmorehouse) [reply src/components/tabs/tabs.js]: As object spread will skip/ignore `null` or `undefined` variables

Comment (andrewvasilchuk) [reply src/components/tabs/tabs.js]: @tmorehouse, thank did not know about it.

File: src/components/tabs/tab.js [hunk 2]
```diff
@@ -39,6 +39,9 @@ export const BTab = /*#__PURE__*/ Vue.extend({
       type: [String, Array, Object],
       default: null
     },
+    titleLinkAttributes: {
+      type: Object
+    },
```
Comment (tmorehouse) [inline src/components/tabs/tab.js]: Add in `default: null` to coerce `undefined` into `null`

Comment (tmorehouse) [reply src/components/tabs/tab.js]: This should be the last tweak needed

Comment (tmorehouse) [reply src/components/tabs/tab.js]: And the `package.json` file in the component's directory will need updating to include this new prop for `BTab` section.

**Pull request merged**